pysat/tests/test_instruments.py

```diff
@@ -84,33 +70,42 @@ def generate_instrument_list(exclude_list, exclude_tags):
                 module._test_dates = info
             for sat_id in info.keys():
                 for tag in info[sat_id].keys():
-                    if name in exclude_tags and \
-                            tag in exclude_tags[name]['tag'] and \
-                            sat_id in exclude_tags[name]['sat_id']:
-                        # we don't want to test download for this combo
+                    try:
+                        inst = pysat.Instrument(inst_module=module,
+                                                tag=tag,
+                                                sat_id=sat_id,
+                                                temporary_file_list=True)
+                        inst._test_dates = module._test_dates
+                        travis_skip = ((os.environ.get('TRAVIS') == 'true')
+                                       and not inst._test_download_travis)
+                        if inst._test_download:
+                            if not travis_skip:
+                                instrument_download.append(inst)
+                        elif not inst._password_req:
+                            # we don't want to test download for this combo
+                            # But we do want to test the download warnings
+                            # for instruments without a password requirement
+                            instrument_no_download.append(inst)
+                    except:
                         pass
-                    else:
-                        try:
-                            inst = pysat.Instrument(inst_module=module,
-                                                    tag=tag,
-                                                    sat_id=sat_id,
-                                                    temporary_file_list=True)
-                            inst._test_dates = module._test_dates
-                            instruments.append(inst)
-                            instrument_modules.append(module)
-                        except:
-                            pass
     pysat.utils.set_data_dir(saved_path, store=False)
 
-    output = {'list': instruments,
-              'names': instrument_names,
-              'modules': instrument_modules}
+    output = {'names': instrument_names,
+              'download': instrument_download,
+              'no_download': instrument_no_download}
 
     return output
 
-instruments = generate_instrument_list(exclude_list, exclude_tags)
+instruments = \
+    generate_instrument_list(instrument_names=pysat.instruments.__all__)
```

Comment:
`instruments = generate_instrument_list(
instrument_names=pysat.instruments.__all__)`
for a cleaner page break